dashboard/views.py

```diff
@@ -66,9 +66,63 @@ def get(self, request):
 
 class WarningsInstance_View(View):
 
-    def get(self, request):
+    def get(self, request, **kwargs):
+
+        cap_id = kwargs['cap_id']
+        cap_instance = CAPInstance.objects.get(cap_id=cap_id)
+
+        # Grab Masterlist Atrributes
+        brokenlist = self.get_broken_urls(cap_instance)
+        retaglist = self.get_retag(cap_instance)
+        nimlist = self.get_nim(cap_instance)
+
+        context = {'date': cap_instance.date, 'brokenlist':brokenlist[:5], 'retaglist': retaglist[:5], 'nimlist': nimlist[:5]}
+
+        return render(request, "warnings/WARNINGS_INSTANCE.html", context)
+
+    def get_broken_urls(self, cap_instance):
+
+        broken_urls_qs = BrokenAPI.objects.filter(cap_id=cap_instance)
+
+        # Get Masterlist Attributes
+        broken_datasets = []
+
+        for broken in broken_urls_qs:
+            masterlist_obj = broken.datagov_ID
```

Comment:
will the `BrokenAPI` instance always have a `datagov_ID`? If not, the code can break.

Reply:
Also same comment for other `get_{metrics}` functions

Reply:
Noted - The cap service will only create a record in this table that associates a masterlist record (datagov_ID) to a capinstance record so yes; however perhaps exception handling is better incase of any database issues?